src/codegen/codegen_compatibility_visitor.cpp

```diff
@@ -50,6 +50,22 @@ std::string CodegenCompatibilityVisitor::return_error_if_solve_method_is_unhandl
     return unhandled_method_error_message.str();
 }
 
+// NOLINTNEXTLINE(readability-convert-member-functions-to-static)
+std::string CodegenCompatibilityVisitor::return_error_extern(
+    ast::Ast& node,
+    const std::shared_ptr<ast::Ast>& ast_node) const {
+    if (simulator == "neuron") {
+        // When generating code for NEURON, NMODL supports EXTERNAL variables,
+        // same as nocmodl.
+        return "";
+    }
+
+    // When generating code for CoreNEURON EXTERNAL variables aren't permitted.
+    auto external = std::dynamic_pointer_cast<ast::External>(ast_node);
+    return fmt::format("Found EXTERN at [{}] while generating code for CoreNEURON.\n",
```

Comment:
Slight nitpick: should be `EXTERNAL` here.

Reply:
fixed, thanks.